orchestra/workflow/directory.py

```diff
@@ -0,0 +1,36 @@
+import json
+import os
+
+from orchestra.workflow.exceptions import WorkflowError
+
+
+def parse_workflow_directory(workflow_directory):
+    parsed = {
+        'versions': [],
+    }
+
+    # Look for and parse the workflow manifest.
+    workflow_files = os.listdir(workflow_directory)
+    if 'workflow.json' not in workflow_files:
+        raise WorkflowError('No "workflow.json" manifest file found.')
+    with open(os.path.join(workflow_directory, 'workflow.json'), 'r') as f:
+        parsed['workflow'] = json.load(f)
+
+    # Look for and parse workflow version subdirectories.
+    workflow_subdirs = [
+        os.path.join(workflow_directory, workflow_file)
+        for workflow_file in workflow_files
+        if os.path.isdir(os.path.join(workflow_directory, workflow_file))]
+    for version_directory in workflow_subdirs:
+        version_files = os.listdir(version_directory)
+        if 'version.json' not in version_files:
+            continue  # Subdirectory wasn't a workflow version.
+
+        with open(os.path.join(version_directory, 'version.json'), 'r') as f:
+            parsed['versions'].append(json.load(f))
+
+    # Complain if the workflow has no versions.
+    if len(parsed['versions']) == 0:
+        raise WorkflowError('Workflow directory {} does not contain any '
```

Comment:
I guess if you just wanted to test that your `workflow.json` parses? Since this code probably shouldn't be called except via `load_workflow()` though, I can't think of a good reason not to say that a versionless workflow is an invalid workflow.